dd-sdk-android-ndk/src/test/cpp/test_generate_backtrace.cpp

```diff
@@ -0,0 +1,20 @@
+#include "test_utils.h"
+
+#include <string>
+#include <regex>
+
+#include "greatest/greatest.h"
+#include "utils/backtrace-handler.h"
+
+TEST test_generate_backtrace(void) {
+    const char *backtrace = backtrace::generate_backtrace();
+    int lines = testutils::count_backtrace_lines(backtrace);
+    // we don't know if the stack is big enough to cover the required max size of 30
+            ASSERT(lines > 0 && lines <= 30);
+            PASS();
```

Comment:
So a backtrace is valid if and only if it has between 0 and 30 lines ? I think you should also check that each line has the proper format too, and possibly check some of the data if possible.

Reply:
Yeah....basically we set the limit to 30 but it can happen to have less than 30 previous command lines when unwinding the stack. I guess I can assert also the lines format and assert at least the line number.